src/webhook/handlers/processImage.js

```diff
@@ -0,0 +1,223 @@
+  POSTBACK_NO_ARTICLE_FOUND,
+} from './utils';
+import gql from 'src/lib/gql';
+import ga from 'src/lib/ga';
+
+export default async function({ data = {} }, event, userId) {
+  const proxyUrl = getLineContentProxyURL(event.messageId);
+  // console.log(`Image url:  ${proxyUrl}`);
+
+  const visitor = ga(userId, '__PROCESS_IMAGE__', proxyUrl);
+  // Track text message type send by user
+  visitor.event({ ec: 'UserInput', ea: 'MessageType', el: event.message.type });
+
+  let replies;
+  data = {
+    // Start a new session
+    sessionId: Date.now(),
+  };
+
+  const {
+    data: { ListArticles },
+  } = await gql`
+    query($mediaUrl: String!) {
+      ListArticles(
+        filter: { mediaUrl: $mediaUrl }
+        orderBy: [{ _score: DESC }]
+        first: 4
+      ) {
+        edges {
+          score
+          node {
+            id
+            articleType
+            attachmentUrl(variant: THUMBNAIL)
+            attachmentHash
+          }
+        }
+      }
+    }
+  `({
+    mediaUrl: proxyUrl,
+  });
+
+  if (ListArticles.edges.length) {
+    // Track if find similar Articles in DB.
+    visitor.event({ ec: 'UserInput', ea: 'ArticleSearch', el: 'ArticleFound' });
+
+    // Track which Article is searched. And set tracking event as non-interactionHit.
+    ListArticles.edges.forEach(edge => {
+      visitor.event({
+        ec: 'Article',
+        ea: 'Search',
+        el: edge.node.id,
+        ni: true,
+      });
+    });
+
+    const articleOptions = ListArticles.edges
+      .map(({ node: { attachmentUrl, id }, score }, index) => {
+        const imgNumber = index + 1;
+        const displayTextWhenChosen = ngettext(
+          msgid`No.${imgNumber}`,
+          `No.${imgNumber}`,
+          imgNumber
+        );
+
+        const similarity = score - 1;
+        const scoreInPercent = Math.floor(similarity * 100);
+        const similarityEmoji = ['😐', '🙂', '😀', '😃', '😄'][
+          Math.floor(similarity * 4.999)
+        ];
+        const looks = ngettext(
+          msgid`Looks ${scoreInPercent}% similar`,
+          `Looks ${scoreInPercent}% similar`,
+          scoreInPercent
+        );
+
+        //show url attachmentUrl
+        return {
+          type: 'bubble',
+          direction: 'ltr',
+          header: {
+            type: 'box',
+            layout: 'horizontal',
+            spacing: 'md',
+            paddingBottom: 'none',
+            contents: [
+              {
+                type: 'text',
+                text: similarityEmoji,
+                flex: 0,
+              },
+              {
+                type: 'text',
+                text: displayTextWhenChosen + ', ' + looks,
+                gravity: 'center',
+                size: 'sm',
+                weight: 'bold',
+                wrap: true,
+                color: '#AAAAAA',
+              },
+            ],
+          },
+          hero: {
+            type: 'image',
+            url: attachmentUrl,
+            size: 'full',
```

Comment:
The height of thumbnail image always does not exceed 240px (set [in rumors-api](https://github.com/cofacts/rumors-api/blob/8eb18c6202be8735e5d7fe31db4fee5acd7eee06/src/graphql/mutations/CreateMediaArticle.js#L67)).
Can we set the [maxHeight](https://developers.line.biz/en/docs/messaging-api/flex-message-layout/#box-max-height) of hero so that images don't get blurry?